---
title: "External Data and Visualisation"
author: "MRC Clinical Sciences Bioinformatics"
date: "29 February 2016"
output: html_document
---

In this practical we will repeat a little from our previous practicals but start working with some external data now.

The external data we use will be Ebf1 ChIP-seq from Mouse genome mm10 (Not mm9!!).

* First lets find the data we need. Look for Ebf Pro-B cell and B cell ChIP-seq Mus Musculus data in AnnotationHub using the **query** function and retrieve into R.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,eval=T)
```

```{r,cache=T}
suppressPackageStartupMessages(
  library(GenomicRanges)
)
suppressPackageStartupMessages(
  library(rGREAT)
)
suppressPackageStartupMessages(
  library(AnnotationHub)
)

library(AnnotationHub)
library(GenomicRanges)
library(rGREAT)

ah = AnnotationHub()

query(ah, c("Ebf", "Mus Musculus"))

ebf_proB <- ah[["AH27957"]]

ebf_B <- ah[["AH27958"]]


```

* Now overlap the two Ebf ChIP-seq datasets to find flattened peaks common to both Pro-B cells and B cells.

```{r,cache=T}
peaksGR_List <- c(ebf_proB,ebf_B)
peaksGR_flat <- reduce(peaksGR_List)
common_Ebf_Peaks <- peaksGR_flat[peaksGR_flat %over% ebf_proB & peaksGR_flat %over% ebf_B ]
```

* Now extract the BioCyc Pathway enrichment results for these peaks using rGREAT.

```{r,cache=T}

seqlevelsStyle(common_Ebf_Peaks) <- "UCSC"

great_Job <- submitGreatJob(common_Ebf_Peaks,species="mm10")
availableCategories(great_Job)

great_ResultTable = getEnrichmentTables(great_Job,category=
                          "Pathway Data")
names(great_ResultTable)
msigdb_great_ResultTable <- great_ResultTable[["MSigDB Pathway"]]

```

*Find denovo motifs underneath flattened peaks common to both Pro B cells and B cells. You will need to install the library for mm10 sequences (BSgenome.Mmusculus.UCSC.mm10.)

```{r,cache=T,eval=F}
seqlevelsStyle(common_Ebf_Peaks) <- "UCSC"
common_Ebf_Peaks_Resized <- resize(common_Ebf_Peaks,200,fix = "center")


genome <- BSgenome.Mmusculus.UCSC.mm10

seqlevelsStyle(common_Ebf_Peaks) <- "UCSC"

common_Ebf_Peaks_Seq <- getSeq(genome,common_Ebf_Peaks)

names(common_Ebf_Peaks_Seq) <- paste0("peak_",seqnames(common_Ebf_Peaks),"_",
                                         start(common_Ebf_Peaks),
                                         "-",
                                         end(common_Ebf_Peaks))

writeXStringSet(common_Ebf_Peaks_Seq,file="common_Ebf_Peaks.fa")
```